src/MongoDB.EntityFrameworkCore/Storage/MongoDatabaseWrapper.cs

```diff
@@ -217,4 +148,148 @@ private static string GetId(IUpdateEntry entry)
             _ => throw new NotSupportedException($"Unknown type for _id property on entity type ${entry.EntityType.DisplayName()}")
         };
     }
+
+    private IEnumerable<MongoUpdate> ConvertUpdateEntriesToMongoUpdates(IEnumerable<IUpdateEntry> entries)
+    {
+        return
+            GetAllChangedRootEntries(entries)
+            .Select(ConvertUpdateEntryToMongoUpdate)
+            .Where(x => x != null)
+            .Cast<MongoUpdate>();
+    }
+
+    private MongoUpdate? ConvertUpdateEntryToMongoUpdate(IUpdateEntry entry)
+    {
+        var collectionName = entry.EntityType.GetCollectionName();
+        var id = GetId(entry);
+
+        return entry.EntityState switch
+        {
+            EntityState.Added => ConvertAddedEntryToMongoUpdate(collectionName, id, entry),
+            EntityState.Deleted => ConvertDeletedEntryToMongoUpdate(collectionName, id, entry),
+            EntityState.Detached => null,
+            EntityState.Modified => ConvertModifiedEntryToMongoUpdate(collectionName, id, entry),
+            EntityState.Unchanged => null,
+            _ => throw new NotSupportedException($"Unexpected entity state: {entry.EntityState}.")
+        };
+    }
+
+    private MongoUpdate ConvertAddedEntryToMongoUpdate(string collectionName, string id, IUpdateEntry entry)
+    {
+        throw new NotImplementedException(); // EF-17
+    }
+
+    private MongoUpdate ConvertDeletedEntryToMongoUpdate(string collectionName, string id, IUpdateEntry entry)
+    {
+        var filter = new BsonDocument("_id", id);
```

Comment:
If we decide to support types beside `string` for the `_id` then we will need to find the correct serializer to serialize the `id` value.

Reply:
Yeah I think we should do that next if you don't want to include it in this piece of work.